### ISSUE TITLE
checkboxList documentation in The Definitive Guide is misleading

### ISSUE THREAD

**Issue by dataskills** [yiisoft/yii2#9908]
I think the following code example (found at http://www.yiiframework.com/doc-2.0/guide-input-forms.html) could be improved:

`echo $form->field($model, 'items[]')->checkboxList(['a' => 'Item A', 'b' => 'Item B', 'c' => 'Item C']);`

In this example there is no need to specify the items attribute as 'items[]' but rather 'items'. As the checkboxList documentation states, "the corresponding submitted value is an array".

I found this out by trying to assign an array to $items before rendering the form but discovered the checkboxes were not selected. This was because the attribute in my $model was $items and (of course) not $items[].

In a nutshell, the example should simply look  like this:

`echo $form->field($model, 'items')->checkboxList(['a' => 'Item A', 'b' => 'Item B', 'c' => 'Item C']);`

**Comment by SilverFire** [comment]
Both of code example give exactly the same result. I've added a test to ensure it's true: https://github.com/yiisoft/yii2/commit/469206d8d09d558c0e948a531581c1cae367e67a

I think that it's better to leave docs without changes because it may confuse developers: why on the line above there are `[]`, and here they're missing?
